src/widget/findonweb.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <QAction>
+#include <QMenu>
+#include <QUrlQuery>
+
+class Track;
+
+class FindOnWeb : public QMenu { // Interface
+    Q_OBJECT
+  public:
+    virtual void addSubmenusForServices(QMenu* pFindOnMenu, const Track& track);
```

Comment:
This is currently a Producer, a special stateless class that produces all the sub menus. 

We can do it that way. Than it shall not inherit menu, just provide a static producer function addSubmenusForServices. 
In this case we need no interface. Inheritance with of static functions does not work. 

The alternative would be to move the code into the constructor and actually inherit QMenu, or an interface class that inherits QMenu itself.

Reply:
> This is currently a Producer, a special stateless class that produces all the sub menus. We can do it that way. Than it shall not inherit menu, just provide a static producer function addSubmenusForServices. In this case we need no interface. Inheritance with of static functions does not work.

I thought in the first place there should be a one super class and it should be over written by their inherited classes. It is
FindOnWeb -> FindOnSoundcloud
FindOnWeb -> FindOnDiscogs
FindOnWeb -> FindOnMusixmatch
I think in that case FindOnWeb should not inherit the menu as you said. All the other sub menus should inherit FindOnWeb and QMenu then. Right?

Factory calls FindOnSoundcloud and Soundcloud will over write the submenu function, openurl function and so on. So at the end I thought adding another service would be easier because we only need to change service name, url's and so on.

Instead of calling addSubmenusForServices in Factory Class, I thought that moving these methods into constructor would be nicer since creating new FindOnSoundcloud() would do everything? Is it possible?

Reply:
Two alternatives: 
* Independent Sub-Factory classes that create plain QMenus with the required things
* QMenu -> FindOnWeb -> FindOnSoundcloud inheritance with the setup code inside the constructor.